tests/unit_test/job_config/fed_job_config_test.py

```diff
@@ -40,3 +41,119 @@ def test_trim_whitespace(self):
         assert expected == job_config._trim_whitespace("site-0, site-1")
         assert expected == job_config._trim_whitespace(" site-0,site-1 ")
         assert expected == job_config._trim_whitespace(" site-0, site-1 ")
+
+    def test_get_args_instance_attributes(self):
+        """Test _get_args with components that have instance attributes in __dict__"""
+
+        class ComponentWithInstanceAttrs:
+            def __init__(self, param1: str = "default1", param2: int = 42):
+                self.param1 = param1
+                self.param2 = param2
+
+        job_config = FedJobConfig(job_name="test_job", min_clients=1)
+        component = ComponentWithInstanceAttrs("custom_value", 100)
+
+        args = job_config._get_args(component, custom_dir=".")
+
+        # Should extract both parameters since they differ from defaults
+        assert args["param1"] == "custom_value"
+        assert args["param2"] == 100
+
+    def test_get_args_instance_attributes_underscore(self):
+        """Test _get_args with components that have properties (not in __dict__)"""
+
+        class ComponentWithProperties:
+            def __init__(self, value: str = "default"):
+                self._value = value
+
+        job_config = FedJobConfig(job_name="test_job", min_clients=1)
+        component = ComponentWithProperties("test_value")
+
+        args = job_config._get_args(component, custom_dir=".")
+
+        assert args["value"] == "test_value"
+
+    def test_get_args_property_not_in_dict(self):
+        """Test _get_args with property attributes that won't appear in __dict__"""
+
+        class ComponentWithProperty:
+            def __init__(self, threshold: float = 0.5):
+                self._threshold = threshold
+
+            @property
+            def threshold(self) -> float:
+                """Property accessible via hasattr but not in __dict__"""
+                return self._threshold * 2  # Different from stored value to test property access
+
+        job_config = FedJobConfig(job_name="test_job", min_clients=1)
+        component = ComponentWithProperty(0.8)
+
+        args = job_config._get_args(component, custom_dir=".")
+
+        # With old __dict__ approach: threshold property would be missed
+        # With new hasattr approach: should find and access the property correctly
+        assert args["threshold"] == 1.6  # 0.8 * 2 from property logic
```

Comment:
This is a problem! 

Your intent is to configure the component with arg threshold = 0.5. When you export the component to job, then the arg becomes 1.6. When this job is loaded and the component constructed, the object's init arg would be 1.6, and its property "threshold" would be 3.2!

Remember that the object you created using job API must be reconstructed precisely when the job is loaded later.

In this particular example, job API won't be able to support it properly.

Reply:
right, this test case should be NOT supported.
Changed implementation to raise Exception